queries/pools/useGetPoolsQuery.ts

```diff
@@ -1,38 +1,41 @@
 import { utils } from 'ethers';
-
 import { getGraphEndpoint } from 'constants/endpoints';
 import { useGetPoolCreateds, PoolCreatedResult } from '../../subgraph';
 import { calculateStatus } from 'utils/time';
-import { NetworkId } from 'constants/networks';
+import { nameToIdMapping, NetworkId } from 'constants/networks';
 
-const useGetPoolsQuery = ({ networkId }: { networkId?: NetworkId }) => {
-	return useGetPoolCreateds(
-		getGraphEndpoint(networkId),
-		{
-			orderBy: 'timestamp',
-			orderDirection: 'desc',
-		},
-		{
-			id: true,
-			name: true,
-			symbol: true,
-			purchaseTokenCap: true,
-			purchaseToken: true,
-			duration: true,
-			sponsorFee: true,
-			sponsor: true,
-			purchaseExpiry: true,
-			purchaseTokenDecimals: true,
-			timestamp: true,
-			poolStatus: true,
-			purchaseDuration: true,
-			contributions: true,
-			dealAddress: true,
-			hasAllowList: true,
-		},
-		{},
-		networkId ? networkId : NetworkId.Mainnet
-	);
+const useGetPoolsQuery = () => {
+	return Object.entries(nameToIdMapping).map(([networkName, networkId]) => ({
+		...useGetPoolCreateds(
```

Comment:
does this still give you an error about when you can call hooks?

Reply:
The only problem I had here was due to the `networkId` not being pass to the other hooks in `subgraph.ts`. After that was solved, all is good